### PR TITLE
Disable adding / editing shadow effects when selected layers have different shadow counts

### PR DESCRIPTION
**Pull request by shaoshing**
This PR is the result of our discussion on adding / editing effects for multiple layers. The conclusion is that adding / editing only works when selected layers have same shadow counts (see Seth's comments below for the cases to consider if want to support any mixed effects). When the counts don't match, the panel displays "Mixed Appearances". Example:

screen shot 2015-07-17 at 5 11 20 pm

Pasting Seth's comments for reference

```
Small scoping this would say if effects do not match, then we do to not allow editing of effects for that selection. That’s kind of annoying, but it is the smallest, simplest thing.

If we want to allow that kind of editing, here’s the  cases I’m thinking about:

1) user selects two layers for the purpose of adding effects to both at the same time.

2) user selects layers with mismatched effects, and changes an effect that is  shared.

3) user selects layers with mismatched effects and changes an effect that is NOT shared.

4) users selects layers with the same effects, but the effects are in different order on each layer. The user then changes an effect value.

5) user selects layers with different effects applied. Adds a new effect to both, and changes values

* I would think for the small scoping scenario: If the effect types match but not the effect's values —it would still work. For instance layers that have a shadow that have different values, but are the same effect order, would be allowed
```

### CODE REVIEW

File: src/js/jsx/sections/style/Shadow.jsx [hunk 1]
```diff
@@ -459,6 +468,13 @@ define(function (require, exports) {
                             type="dropShadow" />
                     );
                 }, this).toList();
+            } else {
+                shadowsContent = (
+                    <div className="shadow-list__list-container__mixed">
+                        <i>Mixed Appearances</i>
```
Comment (volfied) [inline src/js/jsx/sections/style/Shadow.jsx]: This should be localized.

File: src/js/jsx/sections/style/Shadow.jsx [hunk 2]
```diff
@@ -445,10 +445,19 @@ define(function (require, exports) {
             if (layers.isEmpty()) {
                 return null;
             }
-
-            // Group into arrays of dropShadows, by position in each layer
-            var dropShadowGroups = collection.zip(collection.pluck(layers, "dropShadows")),
-                dropShadowList = dropShadowGroups.map(function (dropShadows, index) {
+                
+            var shadowCountsUnmatch = layers.size > 1 &&
+                    Immutable.Set(layers.map(function (l) { return l.dropShadows.size; })).size !== 1,
```
Comment (volfied) [inline src/js/jsx/sections/style/Shadow.jsx]: agh... I mean... there has to be a simpler way for this... or at least can be broken into couple lines to be more readable.

Maybe `layers.some(function (l) { return l.dropShadows.size !== layers.first().dropShadows.size; })`

**Comment by chadrolfs** [comment]
Will this fix #1739?

File: src/js/jsx/sections/style/Shadow.jsx (hunk 2 above)
Comment (shaoshing) [inline src/js/jsx/sections/style/Shadow.jsx]: your suggestion looks great! BTW: it still needs the `layers.size > 1 ` to exclude single layer selection.

**Comment by shaoshing** [comment]
thank you @volfied , it is ready for the second review.

**Comment by volfied** [comment]
LGTM

**Comment by shaoshing** [comment]
thanks :+1:

**Pull request merged**